src/main/java/org/gradle/crypto/checksum/Checksum.java

```diff
@@ -41,11 +43,11 @@
     }
 
     private final ObjectFactory objectFactory;
-    private FileOps fileOps;
+    private final FileOps fileOps;
 
     private FileCollection files;
-    private File outputDir;
-    private Algorithm algorithm;
+    private DirectoryProperty outputDir;
+    private Property<Algorithm> algorithm;
```

Comment:
The properties should be exposed using public getters.
You can then deprecate the existing getter/setters that used plain values.

Reply:
I pushed a new commit applying this strategy.

I noticed something interesting though, when running the tests with Gradle 5.0 I get 
`Caused by: java.lang.NoSuchMethodError: org.gradle.api.file.DirectoryProperty.convention(Lorg/gradle/api/provider/Provider;)Lorg/gradle/api/file/DirectoryProperty;
`
Whereas the `Since` on the `DirectoryProperty` class is `4.3`. 
Please note that this is running fine on `5.1` 
Is there a glitch on the doc or am I missing something @eskatos ?

Reply:
Yep, @Internal is fine.

`DirectoryProperty` is since 4.3 but the error you get is about its `.convention()` function which is since 5.1.
You could use `.set()` instead.

Reply:
agreed, but there is no specific `Since` on the method like this is the case for `files` for instance

Reply:
Yes there is, see

https://github.com/gradle/gradle/blob/344201c165b776ad8f07a21b21942a9780351b0f/subprojects/core-api/src/main/java/org/gradle/api/provider/Property.java#L133-L135

Reply:
We are looking into 2 different locations:
https://docs.gradle.org/current/javadoc/org/gradle/api/file/DirectoryProperty.html#convention-org.gradle.api.provider.Provider-

Reply:
So it seems javadoc doesn't bring the `@since` value to child classes :/